pkg/identity/github/issuer_test.go

```diff
@@ -0,0 +1,176 @@
+			WantErr: true,
+		},
+		`Token missing ref claim should be rejected`: {
+			Token:   `eyJhbGciOiJSUzI1NiIsImtpZCI6Imptczl3WWdPcmxUWjZsSzZTZ2FfaE1JNDZHcXJXYUhicWZfdHJuZ1NxbDAiLCJ0eXAiOiJKV1QifQ.eyJhY3RvciI6ImFpZGFuc3RlZWxlIiwiYXVkIjoic2lnc3RvcmUiLCJiYXNlX3JlZiI6IiIsImV2ZW50X25hbWUiOiJwdXNoIiwiZXhwIjoxLjYzMTY3Mjg1NmUrMDksImhlYWRfcmVmIjoiIiwiaWF0IjoxLjYzMTY3MjU1NmUrMDksImlzcyI6Imh0dHBzOi8vdG9rZW4uYWN0aW9ucy5naXRodWJ1c2VyY29udGVudC5jb20iLCJqb2Jfd29ya2Zsb3dfcmVmIjoiYWlkYW5zdGVlbGUvYXdzLWZlZGVyYXRpb24tZ2l0aHViLWFjdGlvbnMvLmdpdGh1Yi93b3JrZmxvd3MvdGVzdC55bWxAcmVmcy9oZWFkcy9tYWluIiwianRpIjoiOGVhODM3M2UtMGY5ZC00ODlkLWE0ODAtYWMzN2RlZXhhbXBsZSIsIm5iZiI6MS42MzE2NzE5NTZlKzA5LCJyZWZfdHlwZSI6ImJyYW5jaCIsInJlcG9zaXRvcnkiOiJhaWRhbnN0ZWVsZS9hd3MtZmVkZXJhdGlvbi1naXRodWItYWN0aW9ucyIsInJlcG9zaXRvcnlfb3duZXIiOiJhaWRhbnN0ZWVsZSIsInJ1bl9hdHRlbXB0IjoiMSIsInJ1bl9pZCI6IjEyMzU5OTI1ODAiLCJydW5fbnVtYmVyIjoiNSIsInNoYSI6ImJmOTYyNzU0NzFlODNmZjA0Y2U1YzhlYjUxNWMwNGE3NWQ0M2Y4NTQiLCJzdWIiOiJyZXBvOmFpZGFuc3RlZWxlL2F3cy1mZWRlcmF0aW9uLWdpdGh1Yi1hY3Rpb25zOnJlZjpyZWZzL2hlYWRzL21haW4iLCJ3b3JrZmxvdyI6IkNJIn0.BMiMgl8nzJUvuIUdurFo3AufQzwO3v3O1wf_ZPbA4Uv0vp_5CbEPWMH3svtBVJ8y5eaQV7QDQtDWru1SydF8eu7VFI9AdEN6QUTpNgRYpshenUi1mlUmjVOnuYbFhRMEKmocp9C4nXRZsi584a6KhtCAJv4PQ6vIrYh4hklOET_F`,
+			WantErr: true,
+		},
+	}
+
+	issuer := actionsIssuer{
+		IDTokenVerifier: oidc.NewVerifier(
+			`https://token.actions.githubusercontent.com`,
+			&testKeySet{},
+			&oidc.Config{
+				ClientID:        `sigstore`,
+				SkipExpiryCheck: true,
+			},
+		)}
+	ctx := context.TODO()
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			gotPrincipal, err := issuer.Authenticate(ctx, test.Token)
+			if err != nil {
+				if !test.WantErr {
+					t.Error(err)
+				} else {
+					return
+				}
+			}
+			rawPrincipal, ok := gotPrincipal.(*workflowPrincipal)
+			if !ok {
+				t.Fatal("Wrong principal type")
+			}
+
+			if *rawPrincipal != test.ExpectPrincipal {
+				t.Error("oh god")
```

Comment:
nit: error message should be more descriptive

Reply:
Lol yup good catch